test-data/approval-test-generator.ts

```diff
@@ -1,26 +1,51 @@
-import { S3, STS } from 'aws-sdk'
+import { S3, STS, Redshift } from 'aws-sdk'
 import fs = require('fs')
 import { IAM } from '../src/IAM'
 
 const approvalTest = async () => {
     try {        
         // s3Tests( credentials() )
-        stsTests( )
-        stsTests( {externalId: 'Testing'})
+        // stsTests( )
+        // stsTests( {externalId: 'Testing'})
+        redshiftTests(await credentials())
     } catch (err) {
         console.log("Test data generation failed: " + err)
     }
 }
 
-const credentials = async (extraParams?) => {
+const roleNameWithoutAccess = 'beady-eyeDeniedRole'
+
+const credentialsForRole = async (roleName, extraParams?) => {
     let accountId: string = process.env['AWS_ACCOUNT_ID'] || 'unknown'
     let opts = {
-        roleName: 'Obsidian',
+        roleName: roleName,
         accountId: accountId,        
     }    
     return  await IAM.role({...opts,...extraParams}).credentials()
+
 }
 
+const credentials = async (extraParams?) => {
+    let roleName = process.env['AWS_ROLE_NAME'] || 'unknown'
+    return credentialsForRole(roleName,extraParams)
+}
+
+const redshiftTests = async (credentials) => {
+    var redshift = new Redshift({credentials: credentials, region: 'ap-southeast-2'})
+    
+    let existingClusterName = 'test-redshift-cluster'
+    let notExistsClusterName = 'this-does-not-exist'
+    
+    // redshift.describeClusters({ClusterIdentifier: existingClusterName}).promise().then(writeTo('redshift/describeClusters-exists.json'))
+    // redshift.describeClusters({ClusterIdentifier: notExistsClusterName}).promise().catch(writeTo('redshift/describeClusters-not-exists.json'))
```

Comment:
Dead code?

Reply:
It is, but that's because:
1. the approval tests have been hand crafted based on the environment at this point.
2. They need some manual updates to scrub the data once generated.

Therefore, it's here for documentation and uncomment if required at this point.

Will raise separate issue.

Reply:
Issue #12 raised to cover this.